ydb/core/mind/hive/hive_impl.cpp

```diff
@@ -810,14 +810,7 @@ void THive::Handle(TEvPrivate::TEvKickTablet::TPtr &ev) {
     }
 
     BLOG_D("THive::Handle::TEvKickTablet TabletId=" << tabletId);
-    TBestNodeResult result = FindBestNode(*tablet);
-    if (result.BestNode == nullptr) {
-        Execute(CreateRestartTablet(tabletId));
-    } else if (result.BestNode != tablet->Node) {
-        if (IsTabletMoveExpedient(*tablet, *result.BestNode)) {
-            Execute(CreateRestartTablet(tabletId));
-        }
-    }
```

Comment:
The checks here are pointless - we kill the tablet if there is no node for it, but leave it if there is a different node with the move not being expedient. Also the checks did not guarantee  that the tablet would be restarted on a different node - because things might change between this moment and the restart tablet transaction. Since I have to touch this code, removing them seems best - unconditional tablet restart is reasonable for Kick, that's what people use it for.